robot_sdk/SDK/owl_ec_sdk/robot.py

```
@@ -546,12 +603,18 @@ def getStatus(self) -> str:
                 "ALARM",
                 "COLLISION",
             ]
-
-            if self.connect_success:
-                _, result = self.execute("getRobotState")
+            self.connect()
+            try:
+                if not self.connect_success:
+                    raise Exception(f"Robot not connected at IP:{self.ip_address}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)